rosidl_generator_py/resource/_msg.py.em

```diff
@@ -339,13 +339,16 @@ if isinstance(type_, AbstractNestedType):
             field = getattr(self, s)
             fieldstr = repr(field)
             if isinstance(t, rosidl_parser.definition.AbstractSequence):
-                if len(field) == 0:
-                    fieldstr = '[]'
-                else:
-                    assert fieldstr.startswith('array(')
-                    prefix = "array('X', "
-                    suffix = ')'
-                    fieldstr = fieldstr[len(prefix):-len(suffix)]
+                # It is possible that the user assigned a non-array type to
+                # the AbstractSequence field, so only remove 'array(' if it
+                # is in the string.
+                if fieldstr.startswith('array('):
```

Comment:
Please move the check into the `else` block since there is no reason for it on the `if` block.

Reply:
Done in 810ac05